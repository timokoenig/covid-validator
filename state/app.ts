import { entity, persistence } from 'simpler-state'

export type AppState = {
  country: string
  state: string
  purpose: string
  counter: number
  showCounter: boolean
}

export const app = entity<AppState>(
  { country: 'DE', state: '', purpose: '3G', counter: 0, showCounter: false },
  [persistence('app')]
)

export const setCountry = (country: string) => {
  app.set({ ...app.get(), country })
}

export const setState = (state: string) => {
  app.set({ ...app.get(), state })
}

export const setPurpose = (purpose: string) => {
  app.set({ ...app.get(), purpose })
}

export const toggleCounter = () => {
  const state = app.get()
  app.set({ ...state, showCounter: !state.showCounter })
}

export const resetCounter = () => {
  app.set({ ...app.get(), counter: 0 })
}

export const decreaseCounter = () => {
  const state = app.get()
  if (state.counter == 0) return
  app.set({ ...state, counter: state.counter - 1 })
}

export const increaseCounter = () => {
  const state = app.get()
  app.set({ ...state, counter: state.counter + 1 })
}
