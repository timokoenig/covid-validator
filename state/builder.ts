import { entity, persistence } from 'simpler-state'
import { CustomRule } from '../utils/certlogic'

export type BuilderState = {
  customRules: CustomRule[]
}

export const builder = entity<BuilderState>({ customRules: [] }, [persistence('builder')])

export const setRules = (customRules: CustomRule[]) => {
  builder.set({ ...builder.get(), customRules })
}
