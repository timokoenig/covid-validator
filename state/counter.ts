import { entity, persistence } from 'simpler-state'

export const counter = entity(0, [persistence('counter')])

export const reset = () => {
  counter.set(0)
}

export const decrease = () => {
  if (counter.get() == 0) return
  counter.set(counter.get() - 1)
}

export const increase = () => {
  counter.set(counter.get() + 1)
}
