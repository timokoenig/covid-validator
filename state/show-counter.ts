import { entity, persistence } from 'simpler-state'
import { reset } from './counter'

export const showCounter = entity(false, [persistence('show-counter')])

export const toggle = () => {
  reset()
  showCounter.set(!showCounter.get())
}
