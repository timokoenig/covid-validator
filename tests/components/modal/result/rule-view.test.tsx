import React from 'react'
import renderer from 'react-test-renderer'
import { resetAll } from 'simpler-state'
import RuleView from '../../../../components/modal/result/rule-view'
import { app } from '../../../../state/app'

describe('RuleView', () => {
  beforeEach(() => {
    resetAll()
  })

  it('renders correctly', () => {
    app.set({ ...app.get(), country: 'DE', state: 'HH' })

    const tree = renderer.create(<RuleView />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
