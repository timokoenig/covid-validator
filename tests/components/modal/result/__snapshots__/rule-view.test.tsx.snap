// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RuleView renders correctly 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-mfq1v{color:white;font-weight:semibold;}",
      }
    }
    data-emotion="css mfq1v"
  />,
  <span
    className="chakra-text css-mfq1v"
  >
    DE / Hamburg / 3G
  </span>,
]
`;
